src/components/Card/Card.tsx

```diff
@@ -1,17 +1,14 @@
-import React, { useMemo } from 'react';
-import { Link } from 'react-router-dom';
-
-import type { CardProps } from '@types';
-
-import { useGetPokemonDetailsQuery } from '@api/pokemonApiSlice';
-import { usePokemonImage } from '@hooks/usePokemonImage';
-import { useAppDispatch, useAppSelector } from '@store/hooks';
-import { toggleItemSelection } from '@store/slices/selectedItemsSlice';
-import { cn } from '@utils/cn';
+import Link from 'next/link';
```

Comment:
Your Card clearly falls into the client component category.
```suggestion
'use client'

import Link from 'next/link';
```

again, you should use custom Link component